src/Sarif.Converters/FortifyFprConverter.cs

```diff
@@ -0,0 +1,631 @@
+            {
+                using (Stream auditStream = OpenAuditStream(fprArchive))
+                {
+                    ParseAuditStream(auditStream);
+                }
+            }
+        }
+
+        private static Stream OpenAuditStream(ZipArchive fprArchive)
+        {
+            ZipArchiveEntry auditEntry = fprArchive.Entries.Single(e => e.FullName.Equals("audit.fvdl"));
+            return auditEntry.Open();
+        }
+
+        private void ParseAuditStream(Stream auditStream)
+        {
+            var settings = new XmlReaderSettings
+            {
+                DtdProcessing = DtdProcessing.Ignore,
+                IgnoreWhitespace = true,
+                NameTable = _nameTable,
+                XmlResolver = null
+            };
+
+            using (_reader = XmlReader.Create(auditStream, settings))
+            {
+                while (_reader.Read())
+                {
+                    if (AtStartOfNonEmpty(_strings.Uuid))
+                    {
+                        ParseUuid();
+                    }
+                    // Note: CreatedTS is an empty element (it has only attributes),
+                    // so we can't call AtStartOfNonEmpty here.
+                    else if (AtStartOf(_strings.CreatedTimestamp))
+                    {
+                        ParseCreatedTimestamp();
+                    }
+                    else if (AtStartOfNonEmpty(_strings.Build))
+                    {
+                        ParseBuild();
+                    }
+                    else if (AtStartOfNonEmpty(_strings.Vulnerabilities))
+                    {
+                        ParseVulnerabilities();
+                    }
+                    else if (AtStartOfNonEmpty(_strings.Description))
+                    {
+                        ParseRuleFromDescription();
+                    }
+                    else if (AtStartOfNonEmpty(_strings.Snippets))
+                    {
+                        ParseSnippets();
+                    }
+                    else if (AtStartOfNonEmpty(_strings.CommandLine))
+                    {
+                        ParseCommandLine();
+                    }
+                    else if (AtStartOfNonEmpty(_strings.Errors))
+                    {
+                        ParseErrors();
+                    }
+                    else if (AtStartOfNonEmpty(_strings.MachineInfo))
+                    {
+                        ParseMachineInfo();
+                    }
+                }
+            }
+        }
+
+        private void ParseUuid()
+        {
+            _runId = _reader.ReadElementContentAsString();
+        }
+
+        private void ParseCreatedTimestamp()
+        {
+            string date = _reader.GetAttribute(_strings.DateAttribute);
+            string time = _reader.GetAttribute(_strings.TimeAttribute);
+            if (!string.IsNullOrEmpty(date) && !string.IsNullOrEmpty(time))
```

Comment:
>string [](start = 17, length = 6)

Most code in the SDK uses the non-alias actual .NET names for static methods, this is the standard. So, prefer String.Format. NIT